### PR TITLE
New version: Microsoft.WindowsSDK.10.0.22621 version 10.0.22621.755

### LINKED ISSUE #120433: [Package Issue]: `Microsoft.WindowsSDK.10.0.22621`
### Please confirm these before moving forward

- [X] I have searched for my issue and not found a work-in-progress/duplicate/resolved issue.
- [X] I have not been informed if the issue is resolved in a preview version of the winget client.

### Category of the issue

Installation issue.

### Brief description of your issue

`Microsoft.WindowsSDK.10.0.22621` installer fails with exit code `2008`.

### Steps to reproduce

`winget update Microsoft.WindowsSDK.10.0.22621`

### Actual behavior

```
Found Windows Software Development Kit [Microsoft.WindowsSDK.10.0.22621] Version 10.0.22621.2
This application is licensed to you by its owner.
Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
Downloading https://download.microsoft.com/download/7/9/6/7962e9ce-cd69-4574-978c-1202654bd729/windowssdk/winsdksetup.exe
  ██████████████████████████████  1.32 MB / 1.32 MB
Successfully verified installer hash
Starting package uninstall...
Successfully uninstalled
Starting package install...
Installer failed with exit code: 2008
```

### Expected behavior

`Microsoft.WindowsSDK.10.0.22621` successfully updates.

### Environment

```raw
Windows Package Manager v1.5.2201

Windows: Windows.Desktop v10.0.22621.2283
System Architecture: X64
Package: Microsoft.DesktopAppInstaller v1.20.2201.0
```

### Screenshots and Logs

_No response_

### PR DESCRIPTION
**Pull request by CoolPlayLin**
### Pull request has been created with [Komac](https://github.com/russellbanks/Komac) v1.11.0 :rocket:
- Resolve #120433
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/121453)

### CODE REVIEW

**Comment by CoolPlayLin** [comment]
I don't know why it fails

**Comment by stephengillie** [comment]
Automatic Validation ended with:
> 2023-09-30 19:54:27.836 [CLI ] Starting: 'C:\Users\VALIDA~1\AppData\Local\Temp\WinGet\Microsoft.WindowsSDK.10.0.22621.10.0.22621.755\winsdksetup.exe' with arguments '/q /log "C:\Users\Validator\AppData\Local\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\DiagOutputDir\WinGet-Microsoft.WindowsSDK.10.0.22621.10.0.22621.755-2023-09-30-19-54-27.835.log"'
2023-09-30 19:57:09.979 [CLI ] ShellExecute installer failed: 2147942487

**Comment by stephengillie** [comment]
This package installs normally interactively. Is an installer switch missing, possibly to accept something?

**Comment by CoolPlayLin** [comment]
@stephengillie
I don't know anything about this package. Is there any engineer of Microsoft could help out?

**Comment by stephengillie** [comment]
Feel free to reach out to the team that manages this package.

**Comment by CoolPlayLin** [comment]
@stephengillie Do you know who manage this package?

**Comment by CoolPlayLin** [comment]
@just-ero

**Comment by Trenly** [comment]
Stale PR; Please update or close
[Policy] Needs-Author-Feedback

**Comment by BrandonWanHuanSheng** [comment]
I think setup might need to download some additional cab file to install.

**Comment by BrandonWanHuanSheng** [comment]
The root URL is https://download.microsoft.com/download/7/9/6/7962e9ce-cd69-4574-978c-1202654bd729/windowssdk. Currently, you just saw 1 out of 332 files. Because that setup will continue downloading the file while running the setup. Metadata File 
[Metadata.txt](https://github.com/microsoft/winget-pkgs/files/13179065/Metadata.txt)
[Metadata.xlsx](https://github.com/microsoft/winget-pkgs/files/13179068/Metadata.xlsx)
Source file for download that is required to install WindowsSDKSetup.
[Download.txt](https://github.com/microsoft/winget-pkgs/files/13179079/Download.txt)
[Download.xlsx](https://github.com/microsoft/winget-pkgs/files/13179082/Download.xlsx)

**Comment by BrandonWanHuanSheng** [comment]
That setup always responds to commands /? which is help interface.
image
[UnattendedFile.txt](https://github.com/microsoft/winget-pkgs/files/13179455/UnattendedFile.txt)

**Comment by stephengillie** [comment]
Automatic Validation ended with:
> Installation failed with exit code -1978335226 
2023-09-30 19:57:09.979 [CLI ] ShellExecute installer failed: 2147942487

**Comment by stephengillie** [comment]
Manual Validation ends with the package uninstalling. I'm not sure why.

**Comment by BrandonWanHuanSheng** [comment]
I am going to suspect installer file logs.

File: manifests/m/Microsoft/WindowsSDK/10/0/22621/10.0.22621.755/Microsoft.WindowsSDK.10.0.22621.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Created with Komac v1.11.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.5.0.schema.json
+
+PackageIdentifier: Microsoft.WindowsSDK.10.0.22621
+PackageVersion: 10.0.22621.755
+InstallerType: burn
+InstallerSwitches:
+  Silent: /q
+  SilentWithProgress: /q
+UpgradeBehavior: uninstallPrevious
```
Comment (BrandonWanHuanSheng) [inline manifests/m/Microsoft/WindowsSDK/10/0/22621/10.0.22621.755/Microsoft.WindowsSDK.10.0.22621.installer.yaml:10]:
```suggestion
UpgradeBehavior: uninstallPrevious
ElevationRequirement:  elevatesSelf
Scope: machine
```

**Comment by BrandonWanHuanSheng** [comment]
I think the error code is coming out from Scope: user which there is no admin.  So why I requested some change to see if it is work.

**Comment by BrandonWanHuanSheng** [comment]
It look like there is a little bits weird in that pipelines. MSI UpgradeCode failed

**Comment by BrandonWanHuanSheng** [comment]
Is winget problem?
2023-10-28 01:11:58.607 [REPO] Examining ARP entries for User | X64
2023-10-28 01:11:58.828 [CLI ] Installer args: /q /log "C:\Users\Brandon Wan\AppData\Local\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\DiagOutputDir\WinGet-Microsoft.WindowsSDK.10.0.22621.10.0.22621.755-2023-10-28-01-11-58.828.log"
2023-10-28 01:11:58.829 [CLI ] Starting: 'C:\Users\BRANDO~1\AppData\Local\Temp\WinGet\Microsoft.WindowsSDK.10.0.22621.10.0.22621.755\winsdksetup.exe' with arguments '/q /log "C:\Users\Brandon Wan\AppData\Local\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\DiagOutputDir\WinGet-Microsoft.WindowsSDK.10.0.22621.10.0.22621.755-2023-10-28-01-11-58.828.log"'
2023-10-28 01:24:26.665 [CLI ] ShellExecute installer failed: 2147942487
2023-10-28 01:24:26.704 [CLI ] Terminating context: 0x8a150006 at D:\a\_work\1\s\external\pkg\src\AppInstallerCLICore\Workflows\InstallFlow.cpp:1fd

**Comment by CoolPlayLin** [comment]
Any suggestions?

**Comment by BrandonWanHuanSheng** [comment]
It look like there is still having some problem to add this manifest. I will report it to support.microsoft.com  and answers.microsoft.com.

**Comment by BrandonWanHuanSheng** [comment]
This one was able to add into pipelines. https://download.microsoft.com/download/b/8/5/b85bd06f-491c-4c1c-923e-75ce2fe2378e/windowssdk/winsdksetup.exe There is no error issue detected

**Comment by BrandonWanHuanSheng** [comment]
> It look like there is still having some problem to add this manifest. I will report it to support.microsoft.com  and answers.microsoft.com.

https://answers.microsoft.com/en-us/insider/forum/insider_apps-insider_other/it-look-like-winget-silent-install-failed-with-the/410a01a3-734a-4b13-a085-3943969bf641

**Comment by CoolPlayLin** [comment]
Do not close ^^

**Comment by stephengillie** [comment]
I haven't found a way to silently install this package successfully.

**Comment by stephengillie** [comment]
I saw the .NET Framework 4.8.1 was required in the logs. So I tried with `Microsoft.DotNet.Framework.DeveloperPack_4` as a dependency. But the install still failed.

**Comment by BrandonWanHuanSheng** [comment]
Oh, I see. That dependcies is in catalog.

**Comment by CoolPlayLin** [comment]
Do not close ^^ again

**Pull request closed without merging**